service-commands/scripts/userClockValues.js

```diff
@@ -0,0 +1,155 @@
+const { Command } = require('commander')
+
+function commaSeparatedList(value, unusedPrevValue) {
+  return value.split(',')
+}
+
+const program = new Command()
+program
+  .usage('')
+  .option('-h, --handles <handles>', 'Audius handle', commaSeparatedList, [])
+  .option('-i, --user-ids <userIds>', 'Audius user id', commaSeparatedList, [])
+
+// export DISCOVERY_PROVIDER_ENDPOINT="https://discoveryprovider.audius.co"
+const discoveryProviderEndpoint = process.env.DISCOVERY_PROVIDER_ENDPOINT
+
+async function getUserByHandle(handle, discoveryProviderEndpoint) {
+  try {
+    return (
+      await axios({
+        url: `/v1/full/users/handle/${handle}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and handle: ${handle} with ${err}`
+    )
+  }
+}
+
+async function getUserById(userId, discoveryProviderEndpoint) {
+  try {
+    const resp = (
+      await axios({
+        url: `/users?id=${userId}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+
+    if (!resp) {
+      throw new Error(`Failed to find user with userId ${userId}`)
+    }
+
+    return resp
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and user id: ${userId} with ${err}`
+    )
+  }
+}
+async function getClockValues({
+  wallet,
+  creator_node_endpoint: creatorNodeEndpoint
+}) {
+  const primaryCreatorNode = CreatorNode.getPrimary(creatorNodeEndpoint)
+  const secondaryCreatorNodes = CreatorNode.getSecondaries(creatorNodeEndpoint)
+
+  if (!creatorNodeEndpoint) {
+    return {
+      primaryNode: '',
+      primaryClockValue: '',
+      secondaryNodes: [],
+      secondaryClockValues: []
+    }
+  }
+
+  return {
+    primaryNode: primaryCreatorNode,
+    primaryClockValue: await CreatorNode.getClockValue(
+      primaryCreatorNode,
+      wallet
+    ),
+    secondaryNodes: secondaryCreatorNodes,
+    secondaryClockValues: await Promise.all(
+      secondaryCreatorNodes.map(secondaryNode =>
+        CreatorNode.getClockValue(secondaryNode, wallet)
+      )
+    )
+  }
+}
+
+async function run() {
+  try {
+    const { handles, userIds } = parseArgsAndEnv()
+    const userClockValues = await Promise.all([
+      ...handles.map(async handle => {
+        return {
+          ...(await getClockValues(
+            await getUserByHandle(handle, discoveryProviderEndpoint)
+          )),
+          userInfo: `handle: ${handle}`
+        }
+      }),
+      ...userIds.map(async userId => {
+        return {
+          ...(await getClockValues(
+            await getUserById(userId, discoveryProviderEndpoint)
+          )),
+          userInfo: `userId: ${userId}`
+        }
+      })
+    ])
+
+    userClockValues.forEach(
```

Comment:
do you think we can log information from `userClockValues` in the same loop above?

Reply:
Yes, for this case that makes sense.

I think I will split this out into two functions, so that we can call the underlying function when making the call for all users rather than going through a line that has logging.